pytensor/graph/basic.py

```diff
@@ -1522,13 +1522,38 @@ def _compute_deps_cache_(io):
     return rlist
 
 
+def node_toposort(
+    output_nodes: Iterable[Apply] = (),
+    input_nodes: Iterable[Apply] = (),
+) -> Generator[Apply]:
+    computed = {None, *input_nodes}
+    todo = list(output_nodes)
+    while todo:
+        cur = todo[-1]
+        if cur in computed:
+            todo.pop()
+            continue
+        # Since computed includes None we don't need to filter it in this check
+        if all(i.owner in computed for i in cur.inputs):
+            computed.add(cur)
+            yield todo.pop()
+        else:
+            todo.extend(i.owner for i in cur.inputs if i.owner is not None)
+
+
 def io_toposort(
     inputs: Iterable[Variable],
     outputs: Reversible[Variable],
     orderings: dict[Apply, list[Apply]] | None = None,
     clients: dict[Variable, list[Variable]] | None = None,
```

Comment:
We can probably get rid of this clients functionality as it is only used internally in a single Scan rewrite. Users can use `FunctionGraph.clients` to access this property more ergonomically. This further simplifies the role of `general_toposort` as it no longer needs the `deps_cache` argument